### PR TITLE
shar: add page

### PR DESCRIPTION
**Pull request by manny-uncharted**
fixed issue: #9344 

- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/linux/shar.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# shar
+
+> Create a shell archive.
+> More information: <https://manned.org/shar.1>.
```
Comment (navarroaxel) [inline pages/linux/shar.md:4]:
```suggestion
> More information: <https://manned.org/shar>.
```

File: pages/linux/shar.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# shar
```
Comment (navarroaxel) [inline pages/linux/shar.md:1]: If the same command is available in more than 1 platform should be placed in the `common/` directory. Please move.

Comment (manny-uncharted) [reply pages/linux/shar.md]: I've resolved the issue

File: pages/common/shar.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+`shar -p --intermix-type`
+
+- Specify compaction (compression) program:
+
+`shar -C {{program}} --compactor={{program}}`
+
+- Pass level for compression:
+
+`shar -g {{level}} --level-of-compression={{level}}`
```
Comment (kbdharun) [inline pages/common/shar.md:16]:
```suggestion
`shar --level-of-compression={{level}}`
```

File: pages/common/shar.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# shar
+
+> Create a shell archive.
+> More information: <https://manned.org/shar>.
+
+- Specify compression for input files:
+
+`shar -p --intermix-type`
```
Comment (kbdharun) [inline pages/common/shar.md:8]:
```suggestion
`shar --intermix-type`
```

File: pages/common/shar.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+- Specify compaction (compression) program:
+
+`shar -C {{program}} --compactor={{program}}`
```
Comment (kbdharun) [inline pages/common/shar.md:12]:
```suggestion
`shar --compactor={{program}}`
```

File: pages/common/shar.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+- Pass level for compression:
+
+`shar -g {{level}} --level-of-compression={{level}}`
+
+- Decide uuencoding for each file:
+
+`shar -M, --mixed-uuencode`
```
Comment (kbdharun) [inline pages/common/shar.md:20]:
```suggestion
`shar --mixed-uuencode`
```

File: pages/common/shar.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+- Treat all files as binary:
+
+`shar -B, --uuencode`
```
Comment (kbdharun) [inline pages/common/shar.md:24]:
```suggestion
`shar --uuencode`
```

File: pages/common/shar.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+- Treat all files as text:
+
+`shar -T, --text-files`
```
Comment (kbdharun) [inline pages/common/shar.md:28]:
```suggestion
`shar --text-files`
```

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page.

**Review by blueskyson — APPROVED** [review]

File: pages/common/shar.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+- Specify compression for input files:
+
+`shar --intermix-type`
+
+- Specify compaction (compression) program:
+
+`shar --compactor={{program}}`
+
+- Pass level for compression:
```
Comment (kbdharun) [inline pages/common/shar.md:14]:
```suggestion
- Set pass level for compression:
```
How about this?

**Review by kbdharun — APPROVED** [review]

File: pages/common/shar.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+`shar --mixed-uuencode`
+
+- Treat all files as binary:
+
+`shar --uuencode`
+
+- Treat all files as text:
+
+`shar --text-files`
```
Comment (EmilySeville7cfg) [inline pages/common/shar.md:28]:
```suggestion
- Treat all files as binary or text:

`shar --{{uuencode|text-files}}`
```

Optional fix: consecutive examples can be joined.

File: pages/common/shar.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+`shar --intermix-type`
+
+- Specify compaction (compression) program:
+
+`shar --compactor={{program}}`
+
+- Pass level for compression:
+
+`shar --level-of-compression={{level}}`
```
Comment (EmilySeville7cfg) [inline pages/common/shar.md:16]:
```suggestion
`shar --level-of-compression={{1,...,9}}`
```

Optional fix: `{{level}}` can be replaced with valid level number range.

**Review by EmilySeville7cfg — APPROVED** [review]
- No mandatory fixes found, approved.
- This page should be reviewed by `shar` users.
- Thanks for a contribution! 🚀

File: pages/common/shar.md (hunk 11 above)
Comment (kbdharun) [reply pages/common/shar.md:16]: > ```suggestion
> `shar --level-of-compression={{1,...,9}}`
> ```
> 
> Optional fix: `{{level}}` can be replaced with valid level number range.

How about replacing it with 1,2,3,4,5,6,7,8,9 and if there are a lot of numbers we can use a number range.

Comment (EmilySeville7cfg) [reply pages/common/shar.md:16]: > How about replacing it with 1,2,3,4,5,6,7,8,9 and if there are a lot of numbers we can use a number range.

Such format can take place. But IMO it's better to use range based format when we a referring to more than 2 number range.

**Review by pixelcmtd — CHANGES_REQUESTED** [review]
I just compiled GNU shar and tested your examples. All of them don't do anything until I kill them (I assume they're trying to read `stdin`). Can you please make sure that all of your examples actually work?

all of the examples being run and then killed with ctrl-c

**Comment by EmilySeville7cfg** [comment]
> And by the way: This page should go in linux, as other implementations often have different arguments (or none)

Interesting... I guess we can put it in `common/` and present just the [portable options](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax) there (if `shar` works on multiple platforms). But also copy it to `linux/` to show more options.

**Comment by pixelcmtd** [comment]
> I guess we can put it in `common/` and present just the [portable options](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md?rgh-link-date=2022-12-08T07%3A12%3A02Z#option-syntax) there (if `shar` works on multiple platforms).

I just searched through the SUSv4 and apparently `shar` is actually not a POSIX standard, but to quote the macOS man page:

> The `shar` command appeared in 4.4BSD

so it's basically a standard among most Unix-like systems.

The only portable behavior is `shar {{path/to/file1 path/to/file2 ...}}`, which is why I suggested to move it to `linux`, as it differs heavily from the `common` page.

File: pages/common/shar.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
+# shar
+
+> Create a shell archive.
+> More information: <https://manned.org/shar>.
+
+- Specify compression for input files:
```
Comment (navarroaxel) [inline pages/common/shar.md:6]: Sorry, but this example doesn't work. No file is specified so the command expects a list of files from stdin. These are possible variants:
`shar --intermix-type file.sh` or `cat list.txt | shar --intermix-type`

Also the example description doesn't explain what the whole example does.

**Review by navarroaxel — CHANGES_REQUESTED** [review]

**Review by blueskyson — APPROVED** [review]

**Pull request closed without merging**